aten/src/THS/generic/THSTensor.cpp

```diff
@@ -65,218 +51,81 @@ THTensor *THSTensor_(newValues)(const THSTensor *self) {
-  }
-  self->nDimensionI = nDimI;
-  self->nDimensionV = nDimV;
-
-  return self;
+  THError("Internal error! THSTensor_(rawResize)(self, nDimI, nDimV, size) shouldn't be called; use _get_sparse_impl(self)->raw_resize_(dimI, dimV, size) instead");
 }
 
 // directly assign without cloning or retaining (internal method)
 THSTensor* THSTensor_(_move)(THSTensor *self, THLongTensor *indices, THTensor *values) {
-  int empty = THTensor_(nDimension)(values) == 0;
-  if (!empty) {
-    THArgCheck(THLongTensor_nDimension(indices) == 2, 1,
-        "indices must be nDim x nnz");
-    THArgCheck(THLongTensor_size(indices, 1) == THTensor_(size)(values, 0), 1,
-        "indices and values must have same nnz");
-    THArgCheck(THLongTensor_size(indices, 0) == self->nDimensionI, 2,
-        "indices has incorrect first dimension, expected %d, got %d", self->nDimensionI, THLongTensor_size(indices, 0));
-    THArgCheck(THTensor_(nDimension)(values) == self->nDimensionV + 1, 3,
-        "values has incorrect number of dimensions, expected %d, got %d", self->nDimensionV + 1, THTensor_(nDimension)(values));
-  } else {
-    THArgCheck(THLongTensor_nDimension(indices) == 0, 2,
-        "if values is empty, indices must be empty too");
-  }
-  THLongTensor_free(self->indices);
-  THTensor_(free)(self->values);
-  self->indices = indices;
-  self->values = values;
-  self->nnz = empty ? 0 : THTensor_(size)(values, 0);
-  self->coalesced = 0;
-
-  return self;
+  THError("Internal error! THSTensor_(_move)(self, indices, values) shouldn't be called; use _alias_into_sparse(self, indices, values) instead");
 }
 
 THSTensor* THSTensor_(_set)(THSTensor *self, THLongTensor *indices, THTensor *values) {
-  // Note: Not like torch.set, this is an internal method
-  return THSTensor_(_move)(
-    self, THLongTensor_newClone(indices), THTensor_(newClone)(values));
+  THError("Internal error! THSTensor_(_set)(self, indices, values) shouldn't be called; use _copy_into_sparse(self, indices, values) instead");
 }
 
 static inline THSTensor* THSTensor_(_newWithDimsAndTensor)(int64_t nDimI, int64_t nDimV, int64_t *sizes, THLongTensor *indices, THTensor *values) {
-  THSTensor *self = THSTensor_(new)();
-  THSTensor_(rawResize)(self, nDimI, nDimV, sizes);
-
-  // NB: by default, we do NOT clone indices/values into the sparse tensor.
-  // Efficient API by default!
-  THSTensor_(_move)(self, THLongTensor_newWithTensor(indices), THTensor_(newWithTensor)(values));
-  return self;
+  THError("Internal error! THSTensor_(_newWithDimsAndTensor)(nDimI, nDimV, sizes, indices, values) shouldn't be called; use _new_with_dims_and_tensor_sparse(dtype, nDimI, nDimV, sizes, indices, values) instead");
 }
 
 /*** end helper methods ***/
 
 /* Empty init */
 THSTensor *THSTensor_(new)(void)
 {
-  THSTensor *self = (THSTensor *)THAlloc(sizeof(THSTensor));
-  THSTensor_(rawInit)(self);
-  return self;
+  THError("Internal error! THSTensor_(new)() shouldn't be called; use dtype.tensor() instead");
```

Comment:
why did we keep these?  I agree we could eventually have these call ATen -- but what's the point of not just failing at compile time?

Reply:
I kept it so that, if we eventually have these call ATen, all of the templates will still be here so it's an easier patch to do. Otherwise whoever does it has to dig out the old files, wire them all back up, and then start doing replacements.